sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusConnectionEventArgs.cs

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+
+namespace Azure.Messaging.ServiceBus
+{
+    /// <summary>
+    /// The event args specified when the state of the connection with the service changes.
+    /// </summary>
+    public class ServiceBusConnectionEventArgs : EventArgs
+    {
+        /// <inheritdoc cref="ServiceBusClient.FullyQualifiedNamespace"/>
+        public string FullyQualifiedNamespace { get; }
+
+        /// <summary>
+        /// The proxy being used when communicating with the service.
+        /// </summary>
+        public IWebProxy Proxy { get; }
```

Comment:
Proxy and TransportType are interesting additions.    What's the scenario where they'd be useful?   I'd see the client identifier being more helpful to differentiate between multiple instances.

Reply:
Yeah, I considered that but was thinking it might be better to wait until we actually expose Identifier on the client. Mostly, I just wanted to add the event args type now so that we can evolve it in the future.